packages/gitbook/src/app/middleware/(site)/(core)/~gitbook/sitemaps/[[...pathname]]/route.ts

```diff
@@ -0,0 +1,112 @@
+import { RevisionPage, RevisionPageDocument, RevisionPageGroup } from '@gitbook/api';
+import jsontoxml from 'jsontoxml';
+import { NextRequest } from 'next/server';
+
+import { getRevisionPages, getSpace } from '@/lib/api';
+import { getAbsoluteHref, getBasePath } from '@/lib/links';
+import { lookupSiteByAPI } from '@/lib/lookup';
+import { getPagePath } from '@/lib/pages';
+import { joinPath } from '@/lib/paths';
+import { isPageIndexable } from '@/lib/seo';
+
+type Params = {
+    pathname: string[];
+};
+
+export const runtime = 'edge';
+
+/**
+ * Generate a sitemap.xml for the current space.
+ */
+export async function GET(req: NextRequest, { params }: { params: Promise<Params> }) {
+    const { pathname } = await params;
+    const lastPath = pathname[pathname.length - 1];
+    if (lastPath !== 'sitemap.xml') {
+        return new Response('Not found', { status: 404 });
+    }
+    const paths = pathname.slice(0, -1);
+    const basePath = await getBasePath();
+    const origin = `https://${basePath}`;
+    const path = `/${paths.join('/')}`;
```

Comment:
@SamyPesse I have doubt on this logic to build the lookup URL. It works locally but not sure it's the correct approach.

Reply:
Another concern I have is to restrict the ability to generate random sitemaps (one for each page for example).